R/redcap_select_rename.R

```diff
@@ -0,0 +1,191 @@
+#'REDCap Select and Rename
+#'
+#'This function loops through all the variable names of a data set and lets the
+#'user compare them with the variable names set up in REDCap. An API token is
+#'needed to download the variable names from REDCap.Variables with matching
+#'names in REDCap can simply be selected without renaming, variables without
+#'matching names can be selected and renamed. The function returns a data frame
+#'with the selected/renamed variables, writes an overview csv-table, and a short
+#'summary together with the executed code to a log-file for copy-pasting and
+#'adjusting/reusing.
+#'
+#'
+#'@param import_data Data frame to be imported
+#'@param dict Data dictionary (e.g. as
+#'  downloaded from REDCap or via \code{redcap_export_meta(rc_token,
+#'  rc_url)$meta}). If not supplied, this will be downloaded from the API using
+#'  \code{rc_token} and \code{rc_url}.
+#'@param rc_token REDCap API token
+#'@param rc_url Link to REDCap API.
+#'@param forms List of REDCap forms of which variable names will be displayed.
+#'  Default = all forms.
+#'@param auto if TRUE, variables with matching names will be automatically
+#'  selected. If FALSE, the user can decide if the variable shall be imported or
+#'  not. Default = TRUE.
+#'
+#'@return Data frame with selected/renamed data. Log-file with executed code.
+#'  CSV-table with overview.
+#'@export
+#'@importFrom stringr str_detect
+#'@importFrom crayon bold underline blue italic
+#'@importFrom dplyr select filter
+#'
+#' @examples
+#' data(importdemo)
+#' data(meta)
+#' redcap_select_rename(importdemo, meta)
```

Comment:
i wonder if this needs wrapping in `dontrun`... examples are also run when the package is checked, and if the function is waiting for input, i dont know what will happen...

Reply:
(already commented out)